src/spi_flash.c

```diff
@@ -739,9 +739,28 @@ int Drv_SpiInitializeFlash(const FLASHER_SPI_CONFIGURATION_T *ptSpiCfg, FLASHER_
 				/* yes, detected spi flash -> copy all attributes */
 				memcpy(&ptFlash->tAttributes, ptFlashAttr, sizeof(SPIFLASH_ATTRIBUTES_T));
 
+				unsigned short manualSpeed = 0;
+				/* If the passed speed was set manually use the passed value instead of the detected value */
+				if(ptSpiCfg->ulInitialSpeedKhz != 1000 || ptSpiCfg->ulMaximumSpeedKhz != 25000)
+				{
+					manualSpeed = 1;
+					ptFlash->tAttributes.ulClock = ptSpiCfg->ulInitialSpeedKhz;
```

Comment:
I would use ulMaximumSpeedKhz , even though the two values are the same. ulInitialSpeedKhz has benn used during the flash detection.